### PR TITLE
If a user leaves the room, another user joins the same room, and then the user who leaves the room disconnects, the shared data in the room will be deleted.

### PR DESCRIPTION
**Pull request by taketoriokina**
Steps to reproduce(please use [sample for Unity](https://github.com/Cysharp/MagicOnion#streaminghub))
1. Connect as user "A".
2. Join room name "AA" as user "A".
3. User "A" leaves the room name "AA".
4. Connect as user "B".
5. Join room name "AA" as user "B".
6. Disconnect as user "A"(Client-connection dispose).

### CODE REVIEW

**Comment by mayuki** [comment]
Sorry for the late reply.

We have tried to investigate the issue but were not able to reproduce it.
What kind of shared data are you talking about? Do you mean `InMemoryStorage` returned by `AddGroup`?

```csharp
var clientA = new GamingHubClient();
await clientA.ConnectAsync(GrpcChannel.ForAddress("https://localhost:5001/"), "AA", "A"); // (1), (2)
await clientA.LeaveAsync(); // (3)

var clientB = new GamingHubClient();
await clientB.ConnectAsync(GrpcChannel.ForAddress("https://localhost:5001/"), "AA", "B"); // (4), (5)
await clientA.DisposeAsync(); // (6)

... 
storage.AllValues // => [ B ]
```

**Comment by taketoriokina** [comment]
Sorry.

There was a shortage in the steps to reproduce.
Please execute the following additionally.
7. Join room name "AA" as user "C".

User "B" and "C" are in the same room, but there is a difference in shared data(storage).

File: src/MagicOnion.Server/Hubs/Group.ImmutableArray.cs [hunk 1]
```diff
@@ -111,17 +111,20 @@ public ValueTask<bool> RemoveAsync(ServiceContext context)
         {
             lock (gate)
             {
-                members = members.Remove(context);
-                if (inmemoryStorage != null)
+                if (this.members.Contains(context))
```
Comment (mayuki) [inline src/MagicOnion.Server/Hubs/Group.ImmutableArray.cs:114]: Thank you! we have reproduced and confirmed the problem.
It seems that if we skip the process if it is empty, we don't need to check if there is an instance or not, what do you think?

```suggestion
                if (!members.IsEmpty)
```

Comment (taketoriokina) [reply src/MagicOnion.Server/Hubs/Group.ImmutableArray.cs]: Thank you!
I confirmed the operation and there was no problem, so I fixed it.

**Review by mayuki — APPROVED** [review]
Looks good to me!

**Pull request merged**